Source/JavaScriptCore/runtime/ArrayPrototype.cpp

```diff
@@ -577,9 +576,7 @@ JSC_DEFINE_HOST_FUNCTION(arrayProtoFuncToString, (JSGlobalObject* globalObject,
         return JSValue::encode(result);
     }
 
-    JSStringJoiner joiner(globalObject, ","_s, length);
-    RETURN_IF_EXCEPTION(scope, encodedJSValue());
-
+    JSStringJoiner joiner(","_s);
```

Comment:
Same here: what's the reason for not doing `reserveCapacity`?

Reply:
Because Object case can have duplicate JSString in high possibility. So reserveCapacity will end up wasting much.
On the other hand, double / int32 case, will use this buffer pretty much well.

Reply:
Oh right, the object case likely leads to `[Object object]` whereas numbers can different values.  Makes sense.